===================================================================
RCS file: /opt/sourcecast/data/ccvs/repository/gsl/vcl/source/gdi/sallayout.cxx,v
retrieving revision 1.46.10.6
retrieving revision 1.46.10.6.8.1
diff -u -r1.46.10.6 -r1.46.10.6.8.1
--- vcl/source/gdi/sallayout.cxx	2004/04/07 11:02:20	1.46.10.6
+++ vcl/source/gdi/sallayout.cxx	2004/05/03 11:04:49	1.46.10.6.8.1
@@ -822,9 +822,10 @@
                 nXPosMax = nXPos;
         }
 
-        // rightmost cluster edge is leftmost edge of next cluster
+        // when the current cluster overlaps with the next one assume
+        // rightmost cluster edge is the leftmost edge of next cluster
         if( (i > 0) && (nXPosMax > pG[1].maLinearPos.X()) )
-            nXPosMax = pG->maLinearPos.X();
+            nXPosMax = pG[1].maLinearPos.X();
 
         // character width is sum of glyph cluster widths
         pCharWidths[n] += nXPosMax - nXPosMin;
